common/persistence/serialization/interfaces.go

```diff
@@ -23,13 +23,281 @@
+		FailoverNotificationVersion *int64
+		FailoverVersion             *int64
+		ActiveClusterName           *string
+		Clusters                    []string
+		Data                        map[string]string
+		BadBinaries                 []byte
+		BadBinariesEncoding         *string
+		HistoryArchivalStatus       *int16
+		HistoryArchivalURI          *string
+		VisibilityArchivalStatus    *int16
+		VisibilityArchivalURI       *string
+		FailoverEndTime             *time.Time
+		PreviousFailoverVersion     *int64
+		LastUpdatedTime             *time.Time
+	}
+
+	// HistoryBranchRange blob in a serialization agnostic format
+	HistoryBranchRange struct {
+		BranchID    *string
+		BeginNodeID *int64
+		EndNodeID   *int64
+	}
+
+	// HistoryTreeInfo blob in a serialization agnostic format
+	HistoryTreeInfo struct {
+		CreatedTime *time.Time
+		Ancestors   []*HistoryBranchRange
+		Info        *string
+	}
+
+	// WorkflowExecutionInfo blob in a serialization agnostic format
+	WorkflowExecutionInfo struct {
+		ParentDomainID                     *string
+		ParentWorkflowID                   *string
+		ParentRunID                        *string
+		InitiatedID                        *int64
+		CompletionEventBatchID             *int64
+		CompletionEvent                    []byte
+		CompletionEventEncoding            *string
+		TaskList                           *string
+		WorkflowTypeName                   *string
+		WorkflowTimeoutSeconds             *time.Duration
+		DecisionTaskTimeoutSeconds         *time.Duration
+		ExecutionContext                   []byte
+		State                              *int32
+		CloseStatus                        *int32
+		StartVersion                       *int64
+		LastWriteEventID                   *int64
+		LastEventTaskID                    *int64
+		LastFirstEventID                   *int64
+		LastProcessedEvent                 *int64
+		StartTime                          *time.Time
+		LastUpdatedTime                    *time.Time
+		DecisionVersion                    *int64
+		DecisionScheduleID                 *int64
+		DecisionStartedID                  *int64
+		DecisionTimeout                    *time.Duration
+		DecisionAttempt                    *int64
+		DecisionStartedTimestamp           *time.Time
+		DecisionScheduledTimestamp         *time.Time
+		CancelRequested                    *bool
+		DecisionOriginalScheduledTimestamp *time.Time
+		CreateRequestID                    *string
+		DecisionRequestID                  *string
+		CancelRequestID                    *string
+		StickyTaskList                     *string
+		StickyScheduleToStartTimeout       *time.Duration
+		RetryAttempt                       *int64
+		RetryInitialIntervalSeconds        *time.Duration
+		RetryMaximumIntervalSeconds        *time.Duration
+		RetryMaximumAttempts               *int32
+		RetryExpirationSeconds             *time.Duration
+		RetryBackoffCoefficient            *float64
+		RetryExpirationTime                *time.Time
```

Comment:
Just a suggestion, some `time.Time` fields have `Time` suffix while others uses `Timestamp` suffix. Maybe we should unify this?